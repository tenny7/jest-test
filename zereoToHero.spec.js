const pizzas = require('./data.json')

test('test pizza records', () => {
    expect(pizzas).toMatchSnapshot();
    expect(pizzas).toHaveLength(5);
    expect(pizzas.map(pizza => pizza.name)).toEqual([
        'nigerian Pizza',
        'nerican Pizza',
        'Anerican Pizza',
        'bigerian Pizza',
        'cigerian Pizza',
    ]);
});

for (let i = 0; i < pizzas.length; i++) {
    test(`pizzas[${i}] should have properties (id, name, image, desc, price)`, () => {
        expect(pizzas[i]).toHaveProperty('id');
        expect(pizzas[i]).toHaveProperty('name');
        expect(pizzas[i]).toHaveProperty('image');
        expect(pizzas[i]).toHaveProperty('desc');
        expect(pizzas[i]).toHaveProperty('price');
    }); 
}

test('mock implementation of a basic function ', () => {
        const mock = jest.fn(() => 'I am a mock function');
        expect(mock('Calling my mock function!')).toBe('I am a mock function');
        expect(mock).toHaveBeenCalledWith('Calling my mock function!');
});

test('mock test for return value ', () => {
    const mock  = jest.fn();
    mock.mockReturnValueOnce('Hello').mockReturnValueOnce('there!')
    mock()
    mock()
    mock()
    expect(mock).toHaveBeenCalledTimes(3)

    mock('Hello','there','Steve');
    expect(mock).toHaveBeenCalledWith('Hello','there','Steve');
    mock('Steve');
    expect(mock).toHaveBeenLastCalledWith('Steve')
});

test('mock implementation test', () => {
    const mock = jest.fn().mockImplementation(() => 'United Kingdom');
    expect(mock('Location')).toBe('United Kingdom');
    expect(mock).toHaveBeenCalledWith('Location')
});

test('spying on original implementation ', () => {
    const pizza = {
        name: n => `Pizza name: ${n}`,
    }
    const spy = jest.spyOn(pizza,'name')
    expect(pizza.name('Cheese')).toBe('Pizza name: Cheese');
    expect(spy).toHaveBeenCalledWith('Cheese')
});