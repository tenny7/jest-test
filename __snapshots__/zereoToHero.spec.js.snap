// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test pizza records 1`] = `
Array [
  Object {
    "desc": "this is a dummy description of a nigerian Jungle Pizza",
    "id": 1,
    "image": "/image/nigerian.jpg",
    "name": "nigerian Pizza",
    "price": 8,
  },
  Object {
    "desc": "this is a dummy description of a american Jungle Pizza",
    "id": 2,
    "image": "/image/american.jpg",
    "name": "nerican Pizza",
    "price": 9,
  },
  Object {
    "desc": "this is a dummy description of a american Jungle Pizza",
    "id": 2,
    "image": "/image/american.jpg",
    "name": "Anerican Pizza",
    "price": 9,
  },
  Object {
    "desc": "this is a dummy description of a american Jungle Pizza",
    "id": 2,
    "image": "/image/american.jpg",
    "name": "bigerian Pizza",
    "price": 9,
  },
  Object {
    "desc": "this is a dummy description of a american Jungle Pizza",
    "id": 2,
    "image": "/image/american.jpg",
    "name": "cigerian Pizza",
    "price": 9,
  },
]
`;
